kratos/utilities/smp_storage.h

```diff
@@ -0,0 +1,95 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_SMP_STORAGE_H_INCLUDED )
+#define  KRATOS_SMP_STORAGE_H_INCLUDED
+
+// System includes
+#ifdef KRATOS_SMP_OPENMP
+#include <vector>
+#include "utilities/openmp_utils.h"
+#endif
+
+// Project includes
+#include "utilities/parallel_utilities.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+template<class TStorageType>
+class SMPStorage
```

Comment:
```suggestion
class ThreadLocalStorage
```

lets call this what it actually is :)

Reply:
I agree!